pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -265,22 +266,35 @@ func ValidateNodeAffinityArgs(args *config.NodeAffinityArgs) error {
 		return nil
 	}
 	affinity := args.AddedAffinity
-	f := field.NewPath("addedAffinity")
+	path := field.NewPath("addedAffinity")
 	var allErrs field.ErrorList
 	if ns := affinity.RequiredDuringSchedulingIgnoredDuringExecution; ns != nil {
-		_, err := nodeaffinity.NewNodeSelector(ns)
+		_, err := nodeaffinity.NewNodeSelector(ns, nodeaffinity.WithPath(path.Child("requiredDuringSchedulingIgnoredDuringExecution")))
 		if err != nil {
-			// TODO(#96167): Expand all field.Error(s) returned once constructor use them.
-			allErrs = append(allErrs, field.Invalid(f.Child("requiredDuringSchedulingIgnoredDuringExecution"), affinity.RequiredDuringSchedulingIgnoredDuringExecution, err.Error()))
+			allErrs = appendFieldErrorListFromErr(allErrs, err)
 		}
 	}
 	// TODO: Add validation for requiredDuringSchedulingRequiredDuringExecution when it gets added to the API.
 	if terms := affinity.PreferredDuringSchedulingIgnoredDuringExecution; len(terms) != 0 {
-		_, err := nodeaffinity.NewPreferredSchedulingTerms(terms)
+		_, err := nodeaffinity.NewPreferredSchedulingTerms(terms, nodeaffinity.WithPath(path.Child("preferredDuringSchedulingIgnoredDuringExecution")))
 		if err != nil {
-			// TODO(#96167): Expand all field.Error(s) returned once constructor use them.
-			allErrs = append(allErrs, field.Invalid(f.Child("preferredDuringSchedulingIgnoredDuringExecution"), affinity.PreferredDuringSchedulingIgnoredDuringExecution, err.Error()))
+			allErrs = appendFieldErrorListFromErr(allErrs, err)
```

Comment:
could probably just keep appending err as you were, and then flatten the aggregate error at the end, without needing `appendFieldErrorListFromErr` or panics

```
return errors.Flatten(allErrs.ToAggregate())
```

Reply:
Done